### PR TITLE
feat: security migration

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/k8Resource/nodeType/NodeDelete.component.tsx [hunk 1]
```diff
@@ -171,12 +207,13 @@ const NodeDeleteComponent = ({ nodeDetails, appDetails, isDeploymentBlocked }: N
 
             {!!manifestPayload && !!isFELibAvailable && (
                 <SecurityModal
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/k8Resource/nodeType/NodeDelete.component.tsx:209]: Suggestion [PROD]: We should give some reload mechanism on error

File: src/components/app/details/appDetails/appDetails.type.ts [hunk 2]
```diff
@@ -262,12 +261,13 @@ export interface LastUpdatedCardType {
 export interface UseGetAppSecurityDetailsProps {
     appId: string
     envId: string
+    installedAppId: number
     isSecurityScanV2Enabled: boolean
 }
 export interface UseGetAppSecurityDetailsReturnType {
```
Comment (AbhishekA1509) [inline src/components/app/details/appDetails/appDetails.type.ts:267]: Suggestion: Can pick some type?

File: src/components/app/details/appDetails/SecurityVulnerabilityCard.tsx [hunk 3]
```diff
@@ -52,10 +53,11 @@ export const SecurityVulnerabilityCard = ({
         }
     }
 
-    const scanResultSeverities = scanResultResponse?.result.imageScan.vulnerability?.summary.severities // For scan-result Api
+    const scanResultSeverities = scanResultResponse?.result.imageScan.vulnerability?.summary.severities
```
Comment (AbhishekA1509) [inline src/components/app/details/appDetails/SecurityVulnerabilityCard.tsx:56]: Need to abstract

File: src/components/ResourceBrowser/ResourceList/ResourceBrowserActionMenu.tsx [hunk 4]
```diff
@@ -41,9 +68,25 @@ const ResourceBrowserActionMenu: React.FC<ResourceBrowserActionMenuType> = ({
...
+        () =>
+            getResourceScanDetails({
+                name: String(resourceData.name),
+                namespace: String(resourceData.namespace),
+                group: selectedResource?.gvk?.Group,
+                kind: selectedResource?.gvk?.Kind,
+                version: selectedResource?.gvk?.Version,
+                clusterId: +clusterId,
+            }),
+        [],
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/ResourceList/ResourceBrowserActionMenu.tsx:86]: Please verify if showVulnerabilityModal in dep

Comment (arunjaindev) [reply src/components/ResourceBrowser/ResourceList/ResourceBrowserActionMenu.tsx:86]: working fine

File: src/components/security/SecurityScansTab/SecurityScansTab.tsx [hunk 5]
```diff
@@ -65,7 +66,16 @@ export const SecurityScansTab = () => {
         updateSearchParams,
     } = urlFilters
 
-    const isSecurityScanV2Enabled = window._env_.ENABLE_RESOURCE_SCAN_V2 && !!isFELibAvailable
+    const [executionDetailsLoading, executionDetailsResponse, executionsDetailsError] = useAsync(
```
[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/components/security/SecurityScansTab/SecurityScansTab.tsx:69]: added support for v2 as well

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**